### PR TITLE
Update DataCookerPath and DataOutputPath API to not expose internal string rep of paths

### PR DESCRIPTION
**Pull request by mslukebo**
This PR simplifies the API of `DataCookerPath` and `DataOutputPath`. It marks any methods, properties, or constructors exposing the `"[sourceParserId]\[cookerId]\[outputName]"` strings we use as internal representation as obsolete. Additionally, it exposes new methods on these classes:

- `DataCookerPath.ForComposite`
- `DataCookerPath.ForSource`
- `DataOutputPath.ForComposite`
- `DataOutputPath.ForSource`

which users should call instead.

Most files touched in this PR are for updating tests to use the new API.

This PR also updates the migration docs.

### CODE REVIEW

File: src/Microsoft.Performance.SDK/Extensibility/DataCookerPath.cs [hunk 1]
```diff
@@ -92,7 +157,11 @@ public DataCookerPath(DataCookerPath other)
         /// <summary>
         /// Gets the combination of the source parser Id and the data cooker Id.
         /// </summary>
-        public string CookerPath { get; }
+        [Obsolete("This will be removed by SDK v1.0.0 release candidate 1")]
+        public string CookerPath
+        {
+            get;
```
Comment (zachnew-msft) [inline src/Microsoft.Performance.SDK/Extensibility/DataCookerPath.cs:163]: We can go ahead and put this string as a private field and have the now obsolete property and toString return that.

File: src/Microsoft.Performance.SDK/Extensibility/DataOutputPath.cs [hunk 2]
```diff
@@ -115,6 +196,7 @@ public DataOutputPath(DataCookerPath cookerPath, string outputId)
         /// <summary>
         ///     The complete path to the data output.
         /// </summary>
+        [Obsolete("This will be removed by SDK v1.0.0 release candidate 1")]
         public string Path => this.CookerPath.CookerPath + "/" + this.OutputId;
```
Comment (zachnew-msft) [inline src/Microsoft.Performance.SDK/Extensibility/DataOutputPath.cs:200]: See earlier comment about going ahead and factoring out a private field.

File: src/Microsoft.Performance.SDK/Processing/RequiresCompositeCookerAttribute.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+namespace Microsoft.Performance.SDK.Processing
+{
+    /// <summary>
+    /// This attribute is used on a class with static properties that return <see cref="TableDescriptor"/>
+    /// or on individual <see cref="TableDescriptor"/> static properties to indicate that a 
+    /// table described by the <see cref="TableDescriptor"/> requires the identified composite data cooker.
+    /// </summary>
+    [AttributeUsage(System.AttributeTargets.Class, AllowMultiple = true, Inherited = false)]
+    public sealed class RequiresCompositeCookerAttribute
+        : RequiresCookerAttribute
```
Comment (zachnew-msft) [inline src/Microsoft.Performance.SDK/Processing/RequiresCompositeCookerAttribute.cs:16]: Do we still need to inherit this? Could we update the code that finds these to go ahead and look for the new ones, and fall back to the old one if it doesn't find them?

File: src/Microsoft.Performance.SDK/Processing/RequiresSourceCookerAttribute.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+namespace Microsoft.Performance.SDK.Processing
+{
+    /// <summary>
+    /// This attribute is used on a class with static properties that return <see cref="TableDescriptor"/>
+    /// or on individual <see cref="TableDescriptor"/> static properties to indicate that a 
+    /// table described by the <see cref="TableDescriptor"/> requires the identified source data cooker.
+    /// </summary>
+    [AttributeUsage(System.AttributeTargets.Class, AllowMultiple = true, Inherited = false)]
+    public sealed class RequiresSourceCookerAttribute
+        : RequiresCookerAttribute
```
Comment (zachnew-msft) [inline src/Microsoft.Performance.SDK/Processing/RequiresSourceCookerAttribute.cs:16]: See RequiresCompositeCooker

**Review by zachnew-msft — CHANGES_REQUESTED** [review]

File: src/Microsoft.Performance.SDK/Processing/RequiresCompositeCookerAttribute.cs (hunk 3 above)
Comment (mslukebo) [reply src/Microsoft.Performance.SDK/Processing/RequiresCompositeCookerAttribute.cs:16]: Yes but then we're changing our cooker discovery code and will have to remove the legacy finding path once we do RC1. Doing it this way mean it "just works"

File: src/Microsoft.Performance.SDK/Extensibility/DataCookerPath.cs (hunk 1 above)
Comment (jmaxson-ms) [reply src/Microsoft.Performance.SDK/Extensibility/DataCookerPath.cs:163]: [https://github.com/microsoft/microsoft-performance-toolkit-sdk/blob/feature/cookerPaths-delete/src/Microsoft.Performance.SDK/Extensibility/DataCookerPath.cs](https://github.com/microsoft/microsoft-performance-toolkit-sdk/blob/feature/cookerPaths-delete/src/Microsoft.Performance.SDK/Extensibility/DataCookerPath.cs)

**Review by jmaxson-ms — DISMISSED** [review]
:shipit:

File: src/Microsoft.Performance.SDK/Extensibility/DataCookerPath.cs (hunk 1 above)
Comment (mslukebo) [reply src/Microsoft.Performance.SDK/Extensibility/DataCookerPath.cs:163]: yep, Jayson's branch does that. Can I resolve this comment?

Comment (mslukebo) [reply src/Microsoft.Performance.SDK/Extensibility/DataCookerPath.cs:163]: Or do you want me to bring that change over so we can just delete the property

Comment (zachnew-msft) [reply src/Microsoft.Performance.SDK/Extensibility/DataCookerPath.cs:163]: I see the other branch, that's fine.

File: src/Microsoft.Performance.SDK/Processing/RequiresCompositeCookerAttribute.cs (hunk 3 above)
Comment (zachnew-msft) [reply src/Microsoft.Performance.SDK/Processing/RequiresCompositeCookerAttribute.cs:16]: Okay sounds good.

**Review by zachnew-msft — APPROVED** [review]

**Review by jmaxson-ms — APPROVED** [review]

**Review by zachnew-msft — APPROVED** [review]

**Pull request merged**